tests/functional/test_integration.py

```diff
@@ -0,0 +1,12 @@
+import pytest
+from ops.model import ActiveStatus
+from pytest_operator.plugin import OpsTest
+
+@pytest.mark.abort_on_fail
+async def test_build_and_deploy(ops_test: OpsTest):
+    charm = await ops_test.build_charm(".")
+    resources = {
+        "gunicorn-image": "gunicorncharmers/gunicorn-app:20.0.4-20.04_edge"
```

Comment:
You're building and publishing an OCI image as part of the CI workflow here. I think we should use that as part of the integration tests.